src/mjolnir/linkclassification.cc

```diff
@@ -566,8 +566,8 @@ std::vector<uint32_t> GoTowardsIntersection(uint32_t start_node,
     // looking for a non link edge with right direction
     for (const auto& node_edge : bundle.node_edges) {
       const Edge& edge = node_edge.first;
-      if (node_edge.second != prev_edge_idx && !edge.attributes.link &&
-          IsEdgeDriveableInDirection(node, edge, forward)) {
+      if (!edge.attributes.link && IsEdgeDriveableInDirection(node, edge, forward) &&
+          visited_nodes.find(EndNode(node, node_edge.first)) == visited_nodes.end()) {
```

Comment:
changed the order of check to put hash set lookup in the end

Reply:
make sense